### PR TITLE
Cloud Foundry service registry now supports internal routes

### PR DESCRIPTION
**Pull request by christianang**
see commit messages below for context

### CODE REVIEW

**Comment by zachgersh** [comment]
@costinm these were the test changes I was talking about.

**Comment by christianang** [comment]
FYI, it looks like the Cloud Foundry tests were moved into the `test` job. This PR moves it back out because it now needs to run on a machine instead of a docker container. The pilot cloudfoundry test now requires an iptables REDIRECT rule to be setup to redirect a VIP address to Envoy.

File: .circleci/config.yml [hunk 1]
```diff
@@ -220,6 +220,36 @@ jobs:
...
+          at: /go
+      - run: make sync
+      - run:
+          no_output_timeout: 20m
+          when: always
+          command: |
+            export PATH=$GOPATH/bin:$PATH
+            sudo apt update
+            sudo apt install -y iptables
+            sudo iptables -t nat -A OUTPUT -d 127.1.1.1/32 -p tcp -j REDIRECT --to-port 15001
```
Comment (rshriram) [inline .circleci/config.yml]: Can’t this iptables stuff work in containers ?

File: pilot/pkg/serviceregistry/cloudfoundry/servicediscovery.go [hunk 2]
```diff
@@ -101,6 +120,38 @@ func (sd *ServiceDiscovery) Instances(hostname model.Hostname, ports []string, t
...
+	internalRouteServicePort := &model.Port{
+		Port:     sd.ServicePort,
+		Protocol: model.ProtocolTCP,
+		Name:     "tcp",
+	}
+
+	for _, internalRoute := range internalRoutesResp.GetInternalRoutes() {
+		for _, backend := range internalRoute.GetBackends().Backends {
+			if internalRoute.Hostname == hostname.String() {
+				instances = append(instances, &model.ServiceInstance{
```
Comment (rshriram) [inline pilot/pkg/serviceregistry/cloudfoundry/servicediscovery.go]: Don’t you want to dedup with ones added in for loop above?

Comment (christianang) [reply pilot/pkg/serviceregistry/cloudfoundry/servicediscovery.go]: Routes and InternalRoutes are mutually exclusive. This is handled by Cloud Foundry's Copilot, which filters on `apps.internal` domain right now.

**Pull request merged**